craft_application/launchpad/models/recipe.py

```diff
@@ -465,4 +475,33 @@ def _get_lp_recipe(cls, lp: Launchpad) -> Any:
         return lp.lp.charm_recipes
 
 
-Recipe = SnapRecipe | CharmRecipe
+class RockRecipe(_StandardRecipe):
+    """A recipe for a rock.
+
+    https://api.launchpad.net/devel.html#rock_recipe
+    """
+
+    ARTIFACT: ClassVar[Literal["rock"]] = "rock"  # type: ignore[reportIncompatibleVariableOverride]
```

Comment:
What's the benefit of declaring this as `Literal["rock"] = "rock"` over `str = "rock"`? It'd let us remove the type ignore. (Same goes for the charm one up above)

Reply:
it helps with type hinting, specially with method overloads like https://github.com/canonical/craft-application/pull/591/files#diff-ca02b0445f42e189fecddfef8e3f608b7e98ecadeef51d817e11e6c94e863ef1R129